### PR TITLE
Added mine and 2 others.

### PR DESCRIPTION
**Pull request by sara-02**
Added the script too.
Conflicts resolved.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,6 @@
 Let's collect the data of Woman on GitHub, who inspired you to code more, learn more.
 
 **Note:** Please maintain the below list in alphabetical order.
-
-[alicetragedy](https://github.com/alicetragedy)
```
Comment (thefourtheye) [inline README.md:6]: Did you really mean to remove `alicetragedy` and `svaksha`? Why?

File: auto_add_names.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import re
+import os
+
+##part 1- Get the info
+print('Welcome Women Githuber')
+name=input('Enter the name you wish to see on the list::')
+githandle=input('Enter your Git Handle (https://github.com, already appended)::')
+
+f1=open('README.md','r+')
```
Comment (thefourtheye) [inline auto_add_names.py]: Use `with` statement to open files in Python.

File: auto_add_names.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+## This is a small script to automatically add your name in the list alphabetically
+
+import re
+import os
+
+##part 1- Get the info
+print('Welcome Women Githuber')
+name=input('Enter the name you wish to see on the list::')
```
Comment (thefourtheye) [inline auto_add_names.py:13]: This will fail with `NameError` if people use Python 2.x.

File: auto_add_names.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+githandle=input('Enter your Git Handle (https://github.com, already appended)::')
+
+f1=open('README.md','r+')
+f2=open('temp.md','w')
+
+## part 2 -Check the order and insert the values
+for i in range(5):
+	f2.write(next(f1))
```
Comment (thefourtheye) [inline auto_add_names.py]: This is fragile, as it assumes that only the first 5 lines are not names.

File: auto_add_names.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+f1=open('README.md','r+')
+f2=open('temp.md','w')
+
+## part 2 -Check the order and insert the values
+for i in range(5):
+	f2.write(next(f1))	
+for value in f1:
+	line=value
+	if line.strip()!='':
+		line=str(re.findall(r'\[([^]]*)\]', line))
```
Comment (thefourtheye) [inline auto_add_names.py]: This returns a list and you should not convert that to a string like this and process.

File: auto_add_names.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		 	f2.write(line+'\n')
+		 	break
+
+for value in f1:
+	f2.write(value)		 	
+	
+f1.close()
+f2.close()
+os.remove('README.md')
+os.rename('temp.md','README.md')
```
Comment (thefourtheye) [inline auto_add_names.py]: In Python 3, you should prefer [`os.replace`](https://docs.python.org/3/library/os.html#os.replace)

File: README.md (hunk 1 above)
Comment (sara-02) [inline README.md:6]: Rectified the error
err1
err2

File: auto_add_names.py (hunk 4 above)
Comment (sara-02) [inline auto_add_names.py]: Made necessary changes in the code to improve this.

File: auto_add_names.py (hunk 6 above)
Comment (sara-02) [inline auto_add_names.py]: But it does not work with Python2, should I stick to 3 or 2?

File: auto_add_names.py (hunk 3 above)
Comment (sara-02) [inline auto_add_names.py:13]: Yes, it indeed failed, using raw_input it gives error in python3 but works fine in 2.x, what else can be done about it?

Comment (sara-02) [inline auto_add_names.py:13]:
``` python
 if hasattr(__builtins__, 'raw_input'):
      input=raw_input
```

Used this as suggested by Stackoverflow.

File: auto_add_names.py (hunk 5 above)
Comment (sara-02) [inline auto_add_names.py]: What is the alternative then?

File: auto_add_names.py (hunk 3 above)
Comment (thefourtheye) [inline auto_add_names.py:13]: Did you ask a question now? Or refer me to the answered which suggested this?

Comment (sara-02) [inline auto_add_names.py:13]: http://stackoverflow.com/questions/7450223/how-do-i-get-user-input-from-the-keyboard-in-python-2

**Comment by thefourtheye** [comment]
@sara-02 Hmmm, this has to be manually run by people when they submit a patch. How about the system which I proposed in #27?

**Comment by sara-02** [comment]
Can we have a small GUI for this, we can cover the issue #24 by providing 2 separate options/list, one for those with a profile on Github, and other one for private enterprise.

File: README.md (hunk 1 above)
Comment (thefourtheye) [inline README.md:6]: @sara-02 Actually @tapasweni-pathak pointed out that "Why?" might have come off as a little rude. I sincerely apologize. I really didn't mean to sound rude. Hope you understand.

Comment (tapaswenipathak) [inline README.md:6]: No issues @thefourtheye. :+1: 

I know sometimes it becomes hard to control the words and as I said this was to be "extra" polite so this was harder. Avoiding these things create a friendly and comforting environment for the new contributors and that is all, what we all want. :smile: :star:

Comment (thefourtheye) [inline README.md:6]: I totally agree @tapasweni-pathak :-)

Comment (sara-02) [inline README.md:6]: @thefourtheye  no issues :). On a side note I have not pushed the changes in the code and Readme, should I?
or let it stay.

Comment (thefourtheye) [inline README.md:6]: @sara-02 Thanks for understanding :-) Please publish the commits. @tapasweni-pathak can review them once she gets a chance.

**Comment by tapaswenipathak** [comment]
@sara-02 There are merge conflicts can you please correct them?

**Comment by sara-02** [comment]
@tapasweni-pathak done :)

File: auto_add_names.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+## This is a small script to automatically add your name in the list alphabetically
+
+import re
+import os
+
+
+ ## cheking the input function as per python version
+if hasattr(__builtins__, 'raw_input'):
+      input=raw_input
```
Comment (thefourtheye) [inline auto_add_names.py:9]: Use four spaces for Python.

File: auto_add_names.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+##part 1- Get the info
+print('Welcome Women Githuber')
+name=input('Enter the name you wish to see on the list::')
+githandle=input('Enter your Git Handle (https://github.com, already appended)::')
+
+#f1=open('README.md','r+')
```
Comment (thefourtheye) [inline auto_add_names.py:16]: Remove comments

File: auto_add_names.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+#f2=open('temp.md','w')
+
+with open('README.md','r+') as f1, open('temp.md','w') as f2:	
+	for value in f1:
+		line=value
+		if line.strip()!='':
+			line=str(re.findall(r'\[([^]]*)\]', line))
+			line=line.strip('[]\'')
+			if line.lower()<=name.lower():
+				f2.write(value)
```
Comment (thefourtheye) [inline auto_add_names.py:26]: This logic works under the assumption that the names are already in sorted order. I see that most people send PRs by directly editing the page in GitHub.

File: auto_add_names.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+githandle=input('Enter your Git Handle (https://github.com, already appended)::')
+
+#f1=open('README.md','r+')
+#f2=open('temp.md','w')
+
+with open('README.md','r+') as f1, open('temp.md','w') as f2:	
+	for value in f1:
+		line=value
+		if line.strip()!='':
+			line=str(re.findall(r'\[([^]]*)\]', line))
```
Comment (thefourtheye) [inline auto_add_names.py:23]: This still is fragile. I would recommend sorting with the `key` argument, where `key` being a function which returns the captured within `[]`

File: auto_add_names.py (hunk 9 above)
Comment (sara-02) [inline auto_add_names.py:26]: So, I should remove the script altogether?

Comment (thefourtheye) [inline auto_add_names.py:26]: No, it is still helpful for the people who work locally. If @tapasweni-pathak has to merge locally, she also can make use of this :-)

Comment (sara-02) [inline auto_add_names.py:26]: Ok, then will make the improvements, and push the updates.

**Pull request closed without merging**